x/oracle/keeper/keeper.go

```diff
@@ -350,11 +350,9 @@ func (k Keeper) ClearTobinTaxes(ctx sdk.Context) {
 
 // ValidateFeeder return the given feeder is allowed to feed the message or not
 func (k Keeper) ValidateFeeder(ctx sdk.Context, feederAddr sdk.AccAddress, validatorAddr sdk.ValAddress) error {
-	if !feederAddr.Equals(validatorAddr) {
```

Comment:
`sdk.AccAddress` and `sdk.ValAddress` both are []byte types and internally have same bytes except bech32 prefix at stringify. Also Equals function internally check only bytes are same or not, so it is correct implementation.

https://github.com/cosmos/cosmos-sdk/blob/841639486469c2b83103982d6299106283a1b0d4/types/address.go#L177-L183

Reply:
you can check this in test. delegated feeder and origin validator address both are allowed to do feed operation.

https://github.com/terra-money/core/blob/5c709b5b85972806512e4df7b28fe6be8fe95571/x/oracle/keeper/keeper_test.go#L343